scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -937,9 +937,17 @@ class TypedPipeInst[T] private[scalding] (@transient inpipe: Pipe,
       // for us. So unwind until you hit the first filter, snapshot,
       // then apply the unwound functions
       case Some((tap, fields, Converter(conv))) =>
-        Execution.from(new Iterable[T] {
-          def iterator = m.openForRead(conf, tap).asScala.map(tup => conv(tup.selectEntry(fields)))
-        })
+        /*
+           * We create a Stream here so we are lazy with construction, but we don't read twice.
+           */
+        val tryStream = scala.util.Try(m.openForRead(conf, tap)
+          .asScala
+          .map(tup => conv(tup.selectEntry(fields)))
+          .toStream)
```

Comment:
This line here:

https://github.com/twitter/scalding/pull/1230/files#diff-950283e698e7d0a9b2bb04551c277715R60

uses the fact that Stream is traversable to use the Traversable serializer in chill. I tested that in the chill pull request:

https://github.com/twitter/chill/pull/222

I agree it would be good to have some test that fails and this fixes, although it is pretty hard to see how this could possibly still have the read from many machines error.